exonum/src/runtime/dispatcher/mod.rs

```diff
@@ -235,16 +247,17 @@ impl Dispatcher {
             .and_then(|runtime| {
                 runtime.start_service(&spec)?;
                 // Try to configure the started instance of the service, otherwise stop.
-                Self::configure_service(runtime.as_ref(), fork, &spec, constructor).map_err(|e| {
-                    error!(
-                        "An error occurred while configuring the service {}: {}",
-                        spec.name, e
-                    );
-                    if let Err(e) = runtime.stop_service(&spec) {
-                        panic!(FatalError::new(e.to_string()))
-                    }
-                    e
-                })
+                Self::configure_service(runtime.as_ref(), fork, spec.as_descriptor(), constructor)
+                    .map_err(|e| {
+                        error!(
+                            "An error occurred while configuring the service {}: {}",
+                            spec.name, e
+                        );
+                        if let Err(e) = runtime.stop_service(spec.as_descriptor()) {
+                            panic!(FatalError::new(e.to_string()))
```

Comment:
I don't really like this panic. If the service didn't stop correctly, I guess it's better to write an error to the log, and just mark somehow this service as not available.
Service code is client-side and I think it shouldn't be able to crash the core.

Reply:
Ok I add `TODO` to handle this case.